### ISSUE TITLE
[Core feature] [flytekit] Add execution and task project/domain/version to ExecutionParameters

### ISSUE THREAD

**Issue by wild-endeavor** [flyteorg/flyte#2490]
## Goal
Allow access to the task and execution's project, domain, and version in the `ExecutionParameters` object so that users don't have to go through internal environment variables.

## UX
Up for discussion, can create a dataclass with the following fields for example.
```python
ctx = flytekit.current_context()
ctx.identifiers.execution_project
ctx.identifiers.execution_domain
ctx.identifiers.execution_version
ctx.identifiers.execution_name
ctx.identifiers.task_project
ctx.identifiers.task_domain
ctx.identifiers.task_version
ctx.identifiers.task_name
```

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by dylanwilder** [comment]
I don't have a strong opinion as long as the data is there, but how about  
```python
ctx = flytekit.current_context()
ctx.execution.project
ctx.execution.domain
ctx.execution.version
ctx.execution.name
ctx.workflow.project
ctx.workflow.domain
ctx.workflow.version
ctx.workflow.name
ctx.task.project
ctx.task.domain
ctx.task.version
ctx.task.name
```
I could be wrong but this could should be able to reuse already available identifiers?

**Comment by wild-endeavor** [comment]
A few things I came across:
* Executions don't really have a version, so I think for execution the current object should be good enough.
  ```
  flytekit.current_context().execution_id.name  # also .project, .domain
  ```
* We don't expose the workflow attributes nicely... when I ran a task, and looked at the pod, the `FLYTE_INTERNAL_EXECUTION_WORKFLOW` value that came in was `flytesnacks:development:example.wf`, which is fine but I don't really feel like parsing it out from a string unless you feel strong about this.
* Adding a `task_id` property to the `ExecutionParameters` object to mirror the `execution_id` property I think makes sense.
